src/Features/Core/Portable/TodoComments/AbstractTodoCommentService.cs

```diff
@@ -42,9 +58,32 @@ internal abstract class AbstractTodoCommentService : ITodoCommentService
         private async Task<IList<TodoComment>> GetTodoCommentsInRemoteHostAsync(
             RemoteHostClient client, Document document, ImmutableArray<TodoCommentDescriptor> commentDescriptors, CancellationToken cancellationToken)
         {
-            return await client.RunCodeAnalysisServiceOnRemoteHostAsync<IList<TodoComment>>(
-                document.Project.Solution, nameof(IRemoteTodoCommentService.GetTodoCommentsAsync),
-                new object[] { document.Id, commentDescriptors }, cancellationToken).ConfigureAwait(false);
+            var keepAliveSession = await GetKeepAliveSessionAsync(client).ConfigureAwait(false);
```

Comment:
i'm confused by this.  Why are we making a keepalive session, but then not holding onto it.  Won't that mean it will be cleaned up, and will not be 'kept alive'?